src/Mod/Sketcher/Gui/SoZoomTranslation.cpp

```diff
@@ -96,7 +96,7 @@ void SoZoomTranslation::doAction(SoAction * action)
         SbVec3f absVtr = this->abPos.getValue();
         SbVec3f relVtr = this->translation.getValue();
 
-        float sf = this->getScaleFactor(action);
+        float sf = this->calculateScaleFactor(action);
         // For Sketcher Keep Z value the same
         relVtr[0] = (relVtr[0] != 0) ? sf * relVtr[0] : 0;
```

Comment:
This is the same as `relVtr[0] *= sf;`

Reply:
Yes, you are indeed right. But I did not write that line...

I have also seen other things I do not like from this class (such as a hardcoded 5 in calculateScaleFactor). 

I have fought the temptation to make a full review of SoZoomTranslation internal workings at this time. 

Just adding a way to retrieve the scaling factor it uses, because that is what is necessary to fix the actual bug of the issue.

If you have time to do a full review and wish to do it, feel free to make a PR. It is welcome.

Reply:
Maybe eventually. We could create an issue about it, but then again we'd need as many as the number of files qe have!